src/I18nService.js

```diff
@@ -36,39 +36,31 @@ export class I18nService extends EventEmitter {
     this.ensureSupportedLanguage(language)
 
     const dictionary = this.getDictionary(language)
-    if (mark !== undefined && !isNaN(args?.[mark])) {
-      const marked = args?.[mark]
-      let found = null
-      for (let i = 0; i < markedRules.length; i++) {
-        if (
-          dictionary &&
-          dictionary[`${label}.${markedRules[i].suffix}`] &&
-          markedRules[i].condition(marked)
-        ) {
-          found = markedRules[i].suffix
-          break
-        }
-      }
 
-      if (found) {
-        return this.replacePlaceholders(dictionary[`${label}.${found}`], args)
+    const marked = args?.[mark]
+    if (mark && !isNaN(marked)) {
+      const rule = markedRules.find(
+        markedRule =>
+          dictionary?.[`${label}.${markedRule.suffix}`] &&
+          markedRule.condition(marked)
+      )
+
+      if (rule) {
+        return this.replacePlaceholders(
+          dictionary[`${label}.${rule.suffix}`],
+          args
+        )
```

Comment:
I'm not sure this is very readable. Could you do something about it please? Maybe extract something in another function or rename some variables.

Reply:
ab0036f3db4ff2353fa5a23f7286f2df41bfd22d